mage_ai/data_preparation/templates/template.py

```diff
@@ -0,0 +1,140 @@
+from mage_ai.data_cleaner.transformer_actions.constants import (
+    ACTION_CODE_TYPES,
+    ACTION_OPTION_TYPES,
+    OUTPUT_TYPES,
+)
+from mage_ai.data_preparation.models.constants import BlockType
+from mage_ai.data_loader.base import DataSource
+from typing import Mapping, Union
+import jinja2
+import json
+import os
+import shutil
+
+template_env = jinja2.Environment(
+    loader=jinja2.FileSystemLoader('mage_ai/data_preparation/templates')
+)
+
+
+def build_template_from_suggestion(suggestion: Mapping) -> str:
```

Comment:
Do we call this when the suggested_action is in block config?

Reply:
Changed so now if the suggested action payload is passed in `config`, then the custom template is built using that payload.